pkg/api/machine.go

```diff
@@ -37,6 +37,19 @@ type MachineStatus struct {
 	ImageRef               string                   `json:"imageRef"`
 }
 
+func (m *MachineStatus) GetVolumesAsMap() map[string]*VolumeStatus {
+	if m == nil {
```

Comment:
i cannot access to nil pointer in m.VolumeStatus, it will throw panic

Reply:
As far as I know, whether m is a pointer or not, when it is nil, accessing m.VolumeStatus will not cause a panic because nil slices in Go are safe to access. Could you please validate this once?

Reply:
you have right. but m isn't slice. m is pointer to struct which can be nil and i do access to field of struct which is array.

Reply:
make sense. Thank you for the confirmation.